### PR TITLE
package json scan

### PR DESCRIPTION
**Pull request by guyb1**

### CODE REVIEW

File: pkg/oss/oss.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		}
+
+		f_mode := f.Mode()
+
+		if f_mode.IsDir() || !f_mode.IsRegular() {
+			return nil
+		}
+
+		if f.Name() == PACKAGE_JSON_FILE_NAME {
+			packageLockPath := fmt.Sprintf(
```
Comment (VaismanLior) [inline pkg/oss/oss.go:95]: use path.Dir and path.Join instead of working with strings

File: pkg/oss/oss.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		}
+
+		return nil
+	})
+
+	return files
+}
+
+func createPackageJsonFile(dir string, packageJson PackageJson) error {
+	filePath := fmt.Sprintf("%s/%s", dir, PACKAGE_JSON_FILE_NAME)
```
Comment (VaismanLior) [inline pkg/oss/oss.go:111]: path.Join

File: pkg/oss/oss.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+func GeneratePackageLockFiles(path string) (string, map[string]string, error) {
+	files := findPackageJsonFiles(path)
+
+	var (
+		err     error
+		tmpDir  string
+		fileDir string
+	)
+
+	tmpDir, err = ioutil.TempDir(path, "")
```
Comment (VaismanLior) [inline pkg/oss/oss.go:30]: ioutil is deprecated, use os library instead(for all occations)

Comment (VaismanLior) [reply pkg/oss/oss.go:30]: and dont forget to use close where needed

File: pkg/oss/oss.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+	if err != nil {
+		return "", nil, err
+	}
+
+	fileMap := make(map[string]string)
```
Comment (VaismanLior) [inline pkg/oss/oss.go:36]: give it more descriptive name

**Review by tzurielweisberg — APPROVED** [review]

**Pull request merged**